nifi-commons/nifi-vault-utils/src/main/java/org/apache/nifi/vault/config/VaultEnvironment.java

```diff
@@ -0,0 +1,179 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.nifi.vault.config;
+
+import org.apache.nifi.vault.VaultConfigurationException;
+import org.springframework.core.env.Environment;
+import org.springframework.core.env.Profiles;
+
+import java.io.File;
+import java.io.FileReader;
+import java.io.IOException;
+import java.net.URI;
+import java.nio.file.Paths;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.Properties;
+
+/**
+ * A custom Spring Environment that uses configured POJO properties to provide the expected property values
+ * for Spring Vault.
+ */
+public class VaultEnvironment implements Environment {
+    public static final String VAULT_URI = "vault.uri";
+    public static final String VAULT_SSL_KEYSTORE = "vault.ssl.key-store";
+    public static final String VAULT_SSL_KEYSTORE_PASSWORD = "vault.ssl.key-store-password";
+    public static final String VAULT_SSL_KEYSTORE_TYPE = "vault.ssl.key-store-type";
+    public static final String VAULT_SSL_TRUSTSTORE = "vault.ssl.trust-store";
+    public static final String VAULT_SSL_TRUSTSTORE_PASSWORD = "vault.ssl.trust-store-password";
+    public static final String VAULT_SSL_TRUSTSTORE_TYPE = "vault.ssl.trust-store-type";
+    public static final String VAULT_SSL_ENABLED_PROTOCOLS = "vault.ssl.enabled-protocols";
+    public static final String VAULT_SSL_ENABLED_CIPHER_SUITES = "vault.ssl.enabled-cipher-suites";
+
+    private final Properties authProperties;
+    private final VaultProperties vaultProperties;
+
+    public VaultEnvironment(final VaultProperties vaultProperties) throws VaultConfigurationException {
+        Objects.requireNonNull(vaultProperties, "Vault Properties are required");
+        this.vaultProperties = vaultProperties;
+
+        Optional<Properties> authProperties = this.getAuthProperties(vaultProperties);
+        if (!authProperties.isPresent()) {
+            throw new VaultConfigurationException("Vault auth properties file is required");
+        } else {
+            this.authProperties = authProperties.get();
+        }
```

Comment:
This could be adjusted to use `Optional.orElseThrow()`
```suggestion
        this.authProperties = authProperties.orElseThrow(() -> new VaultConfigurationException("Vault auth properties file required"));
```